packages/core/frontend/sirius-components-core/src/modals/impact-analysis/ImpactAnalysisDialog.tsx

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import FiberManualRecordIcon from '@mui/icons-material/FiberManualRecord';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Box from '@mui/material/Box';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import { SxProps, Theme } from '@mui/material/styles';
+import Typography from '@mui/material/Typography';
+import { useEffect, useState } from 'react';
+import { makeStyles } from 'tss-react/mui';
+import { ImpactAnalysisDialogProps } from './ImpactAnalysisDialog.types';
+import { useInvokeImpactAnalysis } from './useImpactAnalysis';
+import { GQLImpactAnalysisReport, ReportViewerProps } from './useImpactAnalysis.types';
+
+const useImpactAnalysisDialogStyles = makeStyles()((theme: Theme) => ({
+  title: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    paddingTop: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+  },
+  titleIcon: {
+    color: theme.palette.primary.main,
+    fontSize: theme.spacing(4),
+  },
+  confirmButton: {
+    backgroundColor: theme.palette.primary.main,
+    color: 'white',
+    '&:hover': {
+      backgroundColor: theme.palette.primary.dark,
+    },
```

Comment:
Use regular variants wether it's for the styling of a text (thanks to typography) or a button. We have a theme on purpose.